src/threadpoolman.cpp

```diff
@@ -84,30 +85,24 @@ void* ThreadPoolMan::Worker(void* arg)
         }
 
         // get instruction
-        std::unique_ptr<thpoolman_param> pparam;
+        thpoolman_param param;
         {
             AutoLock auto_lock(&(psingleton->thread_list_lock));
 
-            if(!psingleton->instruction_list.empty()){
-                pparam = std::move(psingleton->instruction_list.front());
-                psingleton->instruction_list.pop_front();
-                if(!pparam){
-                    S3FS_PRN_WARN("Got a semaphore, but the instruction is empty.");
-                }
+            if(psingleton->instruction_list.empty()){
+                continue;
```

Comment:
I think we need to add the following message from the code before modification here.
It would be nice to have this output when debugging multi-threads.

```
S3FS_PRN_WARN("Got a semaphore, but the instruction is empty.");
```
You can change the message output level from WARN to DBG.

Reply:
Done.